### PR TITLE
(feat) Add Operator Helm Charts

### PR DESCRIPTION
**Pull request by schristoff**
I'll update and add Madhav as an author when merging!

### CODE REVIEW

**Comment by schristoff** [comment]
Should we move over a magefile to publish this somewhere?

**Comment by troy0820** [comment]
This should be published but I wanted to get the grpc server in but that can be in the next iteration.  If we didn't have to move the magefile, let's not ;) but we should publish this. This may not have the latest operator because that needs to be in here as well.

File: helm/charts/operator/crds/apiextensions.k8s.io_v1_customresourcedefinition_porterconfigs.getporter.org.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (troy0820) [inline helm/charts/operator/crds/apiextensions.k8s.io_v1_customresourcedefinition_porterconfigs.getporter.org.yaml:1]: I don't believe we use this anymore.

File: helm/charts/operator/crds/apiextensions.k8s.io_v1_customresourcedefinition_installations.getporter.org.yaml [hunk 2]
```diff
@@ -0,0 +1,229 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (troy0820) [inline helm/charts/operator/crds/apiextensions.k8s.io_v1_customresourcedefinition_installations.getporter.org.yaml:1]: We are also missing the installationoutputs CRD

File: helm/charts/templates/rbac.authorization.k8s.io_v1_clusterrole_porter-operator-manager-role.yaml [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  resources:
+  - credentialsets/status
+  verbs:
+  - get
+  - patch
+  - update
+- apiGroups:
+  - getporter.org
+  resources:
+  - installations
```
Comment (troy0820) [inline helm/charts/templates/rbac.authorization.k8s.io_v1_clusterrole_porter-operator-manager-role.yaml:144]: This is also going to need installationoutputs and it's status as well. 
```yaml
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: porter-operator-manager-role
rules:
- apiGroups:
  - ""
  resources:
  - persistentvolumeclaims
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - ""
  resources:
  - persistentvolumes
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - ""
  resources:
  - secrets
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - ""
  resources:
  - serviceaccounts
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - batch
  resources:
  - jobs
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - getporter.org
  resources:
  - agentactions
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - getporter.org
  resources:
  - agentactions/finalizers
  verbs:
  - update
- apiGroups:
  - getporter.org
  resources:
  - agentactions/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - getporter.org
  resources:
  - agentconfigs
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - getporter.org
  resources:
  - agentconfigs/finalizers
  verbs:
  - update
- apiGroups:
  - getporter.org
  resources:
  - agentconfigs/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - getporter.org
  resources:
  - credentialsets
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - getporter.org
  resources:
  - credentialsets/finalizers
  verbs:
  - update
- apiGroups:
  - getporter.org
  resources:
  - credentialsets/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - getporter.org
  resources:
  - installationoutputs
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - getporter.org
  resources:
  - installationoutputs/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - getporter.org
  resources:
  - installations
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - getporter.org
  resources:
  - installations/finalizers
  verbs:
  - patch
  - update
- apiGroups:
  - getporter.org
  resources:
  - installations/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - getporter.org
  resources:
  - parametersets
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - getporter.org
  resources:
  - parametersets/finalizers
  verbs:
  - update
- apiGroups:
  - getporter.org
  resources:
  - parametersets/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - getporter.org
  resources:
  - porterconfigs
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
```

**Review by troy0820 — APPROVED** [review]
This should work without the suggested changes for the operator as it is today but when releasing the v1.0.0 we need to update this with the grpc server and publish.

**Pull request merged**